emscripten.py

```diff
@@ -1727,9 +1727,11 @@ def create_receiving(function_table_data, function_tables_defs, exported_impleme
     receiving += 'exportAsmFunctions(asm);'
   else:
     if not shared.Settings.SWAPPABLE_ASM_MODULE:
-      if runtime_assertions:
+      if shared.Settings.WASM_ASYNC_COMPILATION and runtime_assertions:
```

Comment:
might be clearer to earlier on, only set `runtime_assertions` if we can, or unset if if we can't?

Reply:
runtime_assertions is used elsewhere in this function below.